vimeo-networking/src/main/java/com/vimeo/networking/model/ProgressiveVideoFile.java

```diff
@@ -0,0 +1,142 @@
+ */
+
+package com.vimeo.networking.model;
+
+import com.google.gson.annotations.SerializedName;
+import com.vimeo.stag.UseStag;
+import com.vimeo.stag.UseStag.FieldOption;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Date;
+
+/**
+ * Representation of a progressive Video stream/playback file.
+ * <p>
+ * Created by zetterstromk on 8/3/17.
+ */
+@SuppressWarnings("unused")
+@UseStag(FieldOption.SERIALIZED_NAME)
+public class ProgressiveVideoFile extends VideoFile {
+
+    private static final long serialVersionUID = 2834083637971280026L;
+
+    @Nullable
+    @SerializedName("type")
+    protected MimeType mMimeType;
+
+    @SerializedName("fps")
+    protected double mFps;
+
+    @SerializedName("width")
+    protected int mWidth;
+
+    @SerializedName("height")
+    protected int mHeight;
+
+    @SerializedName("size")
+    protected long mSize; // size of the file, in bytes
+
+    /** The md5 provides us with a way to uniquely identify video files at {@link #getLink()} */
+    @Nullable
+    @SerializedName("md5")
+    protected String mMd5;
+
+    @Nullable
+    @SerializedName("created_time")
+    protected Date mCreatedTime; // time indicating when this transcode was completed
+
+    public MimeType getType() {
+        return mMimeType == null ? MimeType.NONE : mMimeType;
+    }
+
+    public boolean isVP6() {
+        return getType() == MimeType.VP6;
+    }
+
+    public int getWidth() {
+        return mWidth;
+    }
+
+    public int getHeight() {
+        return mHeight;
+    }
+
+    public long getSize() {
+        return mSize;
+    }
+
+    public double getFps() {
+        return mFps;
+    }
+
+    @Nullable
+    public String getMd5() {
+        return mMd5;
+    }
+
+    @Nullable
+    public Date getCreatedTime() {
+        return mCreatedTime;
+    }
+
+    private static final int HEIGHT_720P = 720;
+    private static final int WIDTH_720P = 1280;
+
+    @NotNull
+    @Override
+    public VideoQuality getQuality() {
+        if (mHeight > HEIGHT_720P || mWidth > WIDTH_720P) {
+            return VideoQuality.HD;
+        } else {
+            return VideoQuality.SD;
+        }
+    }
+
+    // -----------------------------------------------------------------------------------------------------
+    // Equals/Hashcode
+    // -----------------------------------------------------------------------------------------------------
+    // <editor-fold desc="Equals/Hashcode">
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || !(o instanceof ProgressiveVideoFile)) {
+            return false;
+        }
+
+        ProgressiveVideoFile that = (ProgressiveVideoFile) o;
+
+        return (mLink != null && that.getLink() != null) && mLink.equals(that.getLink());
+    }
+
+    @Override
+    public int hashCode() {
+        return mLink != null ? mLink.hashCode() : 0;
```

Comment:
Actually, since they all just compare the `link` value the subclasses do not need a full implementations of `equals` and `hashCode`. 

`equals` can keep the `if (o == null || !(o instanceof ProgressiveVideoFile)) { return false; }` check then just do a `return super.equals(o)`.  

`hashCode()` could either be removed from these 3 subclasses or you could just `return super.hashCode()` if you want to callout the fact that there is a custom implementation in `VideoFile`.